### PR TITLE
Change to npm rather than yarn

### PR DESCRIPTION
**Pull request by bameyrick**
### What is the context of this PR?
Having discussed with @boxadesign I've converted the project to use npm package-lock.json rather than yarn, as this allows us to use tools such as `npm audit`.

I've also:
* removed unused dependencies
* upgraded any packages with vulnerabilities higher than low
* added `upgrade-interactive` script

### CODE REVIEW

**Comment by saedwards** [comment]
Why don't we use https://yarnpkg.com/lang/en/docs/cli/audit/?

**Comment by bameyrick** [comment]
@saedwards I didn't realise they had released yarn audit yet... Now that they've released it it makes converting to npm less compelling as the only other benefits are npm version changing with nvm and npx, although we could still use npx with yarn.

I'm happy to revoke this PR and removal of unused dependencies / upgrading of 
packages with vulnerabilities with yarn if everyone thinks this is the best option?

**Comment by bameyrick** [comment]
I would say it would be less conflicts definitely. The reason for switching back was the lack of an audit feature in yarn but now that it has it there's not much of a reason to change.

Personally I prefer yarn just because I'm too lazy to type `run` ¯\_(ツ)_/¯

Currently pros and cons I'm aware of are:

### Yarn
**Pros**
* Nicer CLI
* Workspaces
* Offline install (if all packages are cached)
* `upgrade-interactive`

**Cons**
* Only uses SHA1 integrity not SHA256

### npm
**Pros**
* Faster install (depends though, for some repos yarn is faster)
* Automatic vulnerability reporting on install

**Cons**
* Noisy CLI
* Have to type `run`

**Comment by boxadesign** [comment]
@bameyrick If we are going to stick with `yarn` can we stop the `package.lock` file being generated or causing the warning on this branch so we can merge it in?

File: docs/developer/03-fractal.md [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
 ---
 
-### Yarn & Gulp
+### npm & Gulp
 
 Commands available in the CLI.
 
-- `yarn start` - start local instance of the fractal build
-- `yarn compile_assets` - compiles all local assets, .js and .css
+- `npm run start` - start local instance of the fractal build
```
Comment (owencjones) [inline docs/developer/03-fractal.md]: In NPM, you don't have to type `run` for the `start` task: https://docs.npmjs.com/cli/start - provided it's a task within the `scripts` section of the package.json

**Comment by bameyrick** [comment]
> @bameyrick If we are going to stick with `yarn` can we stop the `package-lock.json` file being generated or causing the warning on this branch so we can merge it in?

I can use `yarn import` to generate a `yarn.lock` file from the `package-lock.json` and then delete the  `package-lock.json`, and then obviously change all the docs back to yarn commands. 

Once conflicts are resolved this could be merged in then.

File: docs/developer/03-fractal.md (hunk 1 above)
Comment (owencjones) [reply docs/developer/03-fractal.md]: AFAIK, so for at least 5 years, `start`, `stop` and `test` have always been inbuilt scripts.  I think `prestart` etc. were added more recently.

Comment (owencjones) [reply docs/developer/03-fractal.md]: `npm run start` still works, of course

Comment (bameyrick) [reply docs/developer/03-fractal.md]: Another classic _Ben should read the docs™_ moment
 ლ(ಠ_ಠ ლ

**Comment by bameyrick** [comment]
I've put everything back to using yarn now, but kept the vulnerability fixes, package.json sort so this would be ok to merge if everyone is happy

**Pull request merged**